gosling/display.py

```diff
@@ -102,36 +103,13 @@ def __call__(self, spec: GoslingSpec, **meta: Any):
         html = spec_to_html(spec=spec, output_div=self.output_div, **kwargs)
         return {"text/html": html}
 
-
-html_renderer = HTMLRenderer()
+renderers = PluginRegistry[Renderer]("gosling.renderer")
+renderers.register("default", HTMLRenderer())
+renderers.enable("default")
 
 CustomTheme = Dict[str, Any]
 
-
-@dataclass
-class ThemesRegistry:
-    themes: Set[str]
-    custom_themes: Dict[str, CustomTheme] = field(default_factory=dict)
-    active: Optional[str] = None
-
-    def register(self, name: str, theme: CustomTheme) -> None:
-        assert (
-            name not in self.themes
-        ), f"cannot override built-in themes, {self.themes}"
-        self.custom_themes[name] = theme
-
-    def enable(self, name: str) -> None:
-        assert (
-            name in self.custom_themes or name in self.themes
-        ), f"theme must be one of {self.themes} or {set(self.custom_themes.keys())}."
-        self.active = name
-
-    def get(self) -> Union[None, str, CustomTheme]:
-        if self.active is None:
-            return None
-        if self.active in self.themes:
-            return self.active
-        return self.custom_themes[self.active]
-
-
-themes = ThemesRegistry(THEMES)
+themes = PluginRegistry[CustomTheme]("gosling.theme")
+for theme in THEMES:
+    # Add builtin string themes
+    themes.register(theme, theme) # type: ignore
```

Comment:
we could probably also add a default theme

```py
themes.register('default', '') # or themes.register('default', 'light')
```

consistent with renderers.

Reply:
By default it is `None` so that it defers to Gosling.js. Since this isn't a required field, i think None default is ok.